transports/websocket-websys/src/lib.rs

```diff
@@ -90,6 +102,12 @@ impl libp2p_core::Transport for Transport {
         let url = extract_websocket_url(&addr)
             .ok_or_else(|| TransportError::MultiaddrNotSupported(addr))?;
 
+        if !WebContext::is_websocket_supported() {
+            panic!(
+                "Websockets are not supported in this environment, you can use the `isomorphic-ws` npm package to polyfill `global.WebSocket` to enable support."
+            );
+        }
```

Comment:
We can't panic here, this is bad library behaviour! I'd suggest that instead of interacting with `WebSocket` directly here, we create an API on `WebContext` that makes a new connection, something like:

```rust
impl WebContext {
	fn dial(&self, url: String) -> Result<Connection, ...> { }
}
```

Note the `&self`. This forces the `transport` module to go through `WebContext::new` which already checks that websockets are supported in this environment. If it returns `None` you can return an `Error` here that fails the `DialFuture` with a message that websockets are not supported.